### PR TITLE
Raneen Alzafarani

### PR DESCRIPTION
**Pull request by Raneen-z**

### CODE REVIEW

File: PizzaBuilder/PizzaBuilder/Program.cs [hunk 1]
```diff
@@ -0,0 +1,176 @@
...
+
+            var whitePizzaBuilder = new WhitePizzaBuilder();
+            director.Builder = whitePizzaBuilder;
+            //Wheat Pizza With Suase
+            director.WithSuase();
+            Console.WriteLine(whitePizzaBuilder.GetPizza().listParts());
+            //Wheat Pizza With Suase
+            director.WithoutSuase();
+            Console.WriteLine(whitePizzaBuilder.GetPizza().listParts());
+            //Bonus 1
```
Comment (nm-safcsp) [inline PizzaBuilder/PizzaBuilder/Program.cs:170]: Good job doing the first Bonus

File: PizzaBuilder/PizzaBuilder/Program.cs [hunk 2]
```diff
@@ -0,0 +1,176 @@
...
+            this._builder.BuildDough();
+            this._builder.BuildSauce();
+        }
+    }
+    class Program
+    {
+        static void Main(string[] args)
+        {
+            var director = new Director();
+            var wheatPizzaBuilder = new WheatPizzaBuilder();
```
Comment (nm-safcsp) [inline PizzaBuilder/PizzaBuilder/Program.cs:148]: It would be better if you used polymorphism here, without declearing two builders.

**Review by nm-safcsp — COMMENTED** [review]
**Good job Raneen!** 
Well done completing the project and doing the first Bonus.
> don't forget to check comments on your code.

**Comment by nm-safcsp** [comment]
**Good job Raneen!** 
Well done completing the project and doing the first Bonus.
> don't forget to check comments on your code.

File: PizzaBuilder/PizzaBuilder/Program.cs (hunk 2 above)
Comment (Raneen-z) [reply PizzaBuilder/PizzaBuilder/Program.cs:148]: How will the director then know what type of builder I want ?

Comment (nm-safcsp) [reply PizzaBuilder/PizzaBuilder/Program.cs:148]: If you want , you can apply casting on the object to the desired builder.. you can do it this way: 
```C#
 var director = new Director();
            //var wheatPizzaBuilder = new WheatPizzaBuilder();
            IPizzaBuilder pizzaBuilder = new WheatPizzaBuilder();
            director.Builder = (WheatPizzaBuilder)pizzaBuilder;

            //Wheat Pizza With Suase
            director.WithSuase();
            Console.WriteLine(((WheatPizzaBuilder)pizzaBuilder ).GetPizza().listParts());
            //Wheat Pizza With Suase
            director.WithoutSuase();
            Console.WriteLine(((WheatPizzaBuilder) pizzaBuilder ).GetPizza().listParts());
            //Bonus 1 
            pizzaBuilder.BuildDough();
            pizzaBuilder.BuildTopping();
            Console.WriteLine(((WheatPizzaBuilder) pizzaBuilder ).GetPizza().listParts());


            // var whitePizzaBuilder = new WhitePizzaBuilder();
            director.Builder = (pizzaBuilder = new WhitePizzaBuilder());

            director.Builder = pizzaBuilder;
            //Wheat Pizza With Suase
            director.WithSuase();
            Console.WriteLine(((WhitePizzaBuilder)pizzaBuilder ).GetPizza().listParts());
            //Wheat Pizza With Suase
            director.WithoutSuase();
            Console.WriteLine(((WhitePizzaBuilder) pizzaBuilder ).GetPizza().listParts());
            //Bonus 1 
            pizzaBuilder.BuildDough();
            pizzaBuilder.BuildTopping();
            Console.WriteLine(((WhitePizzaBuilder) pizzaBuilder).GetPizza().listParts());
```
or this way:
```C#
var director = new Director();
            //var wheatPizzaBuilder = new WheatPizzaBuilder();
            IPizzaBuilder pizzaBuilder = new WheatPizzaBuilder();
            director.Builder = pizzaBuilder;

            //Wheat Pizza With Suase
            director.WithSuase();
            Console.WriteLine((pizzaBuilder as WheatPizzaBuilder).GetPizza().listParts());
            //Wheat Pizza With Suase
            director.WithoutSuase();
            Console.WriteLine((pizzaBuilder as WheatPizzaBuilder).GetPizza().listParts());
            //Bonus 1 
            pizzaBuilder.BuildDough();
            pizzaBuilder.BuildTopping();
            Console.WriteLine((pizzaBuilder as WheatPizzaBuilder).GetPizza().listParts());


            // var whitePizzaBuilder = new WhitePizzaBuilder();
            director.Builder = (pizzaBuilder = new WhitePizzaBuilder());

            director.Builder = pizzaBuilder;
            //Wheat Pizza With Suase
            director.WithSuase();
            Console.WriteLine((pizzaBuilder as WhitePizzaBuilder).GetPizza().listParts());
            //Wheat Pizza With Suase
            director.WithoutSuase();
            Console.WriteLine((pizzaBuilder as WhitePizzaBuilder).GetPizza().listParts());
            //Bonus 1 
            pizzaBuilder.BuildDough();
            pizzaBuilder.BuildTopping();
            Console.WriteLine((pizzaBuilder as WhitePizzaBuilder).GetPizza().listParts());
```
I hope I answered your question! Thank you.

**Pull request closed without merging**